### PR TITLE
ssh options env var

### PR DESCRIPTION
**Pull request by andrevtg**
Proposal: DOCKER_HOST_SSH_OPTIONS environment variable

**- What I did**
The new "ssh://" protocol for DOCKER_HOST is great, but sometimes we need to use different ssh options than the default. A common scenario if the need to point to a project-specific ssh config file instead of the default one at "~/.ssh/config".

**- How I did it**
An environment variable DOCKER_HOST_SSH_OPTIONS that can hold options for the ssh command docker uses when DOCKER_HOST contains the "ssh://" prefix.

**- How to verify it**
export DOCKER_HOST_SSH_OPTIONS="-F /my/own/ssh.cfg"
export DOCKER_HOST="ssh://somehost"
docker version

**- Description for the changelog**
Just a couple lines at ssh.go, no big deal.

**- A picture of a cute animal (not mandatory but encouraged)**

                    ##        .            
              ## ## ##       ==            
           ## ## ## ##      ===            
       /""""""""""""""""\___/ ===        
  ~~~ {~~ ~~~~ ~~~ ~~~~ ~~ ~ /  ===- ~~~   
       \______ o          __/            
         \    \        __/             
          \____\______/

### CODE REVIEW

File: cli/connhelper/ssh/ssh.go [hunk 1]
```diff
@@ -4,10 +4,15 @@ package ssh
...
 import (
 	"net/url"
+	"os"
 
 	"github.com/pkg/errors"
 )
 
+var (
+	dockerSSHOptions = os.Getenv("DOCKER_SSH_OPTIONS")
+)
```
Comment (AkihiroSuda) [inline cli/connhelper/ssh/ssh.go]: DOCKER_HOST_SSH_OPTIONS, because it has nothibg to do with docker build --ssh

File: cli/connhelper/ssh/ssh.go [hunk 2]
```diff
@@ -65,6 +72,9 @@ func (sp *sshSpec) Args() []string {
 	if sp.port != "" {
 		args = append(args, "-p", sp.port)
 	}
+	if sp.options != "" {
+		args = append(args, sp.options)
```
Comment (AkihiroSuda) [inline cli/connhelper/ssh/ssh.go]: needs to split?

File: cli/connhelper/ssh/ssh.go (hunk 1 above)
Comment (andrevtg) [reply cli/connhelper/ssh/ssh.go]: Changed var name & split string.

**Comment by thaJeztah** [comment]
@andrevtg can you squash your commits, and check if the squashed commit has a correct sign-off?

Also looks like there's a linting failure;

```
cli/connhelper/ssh/ssh.go:1::warning: file is not gofmted with -s (gofmt)
cli/connhelper/ssh/ssh.go:1::warning: file is not goimported (goimports)
```

**Comment by andrevtg** [comment]
Git has special ways to make me feel stupid. I just screwed my last commit when trying to ammend it.

Oh boy.

**Pull request closed without merging**